### ISSUE TITLE
JSONFormatter should be able to customize `msg` field name

### ISSUE THREAD

**Issue by dqminh** [sirupsen/logrus#367]
Our log parsing infrastructure currently assumes that full-text message have `message` name ( similar to logstash json format )

```
{
  "message"    => "hello world",
  "@version"   => "1",
  "@timestamp" => "2014-04-22T23:03:14.111Z",
  "type"       => "stdin",
  "host"       => "hello.local"
}
```

However the current json formatter assumes that the field is called `msg` without any option to modified ( https://github.com/Sirupsen/logrus/blob/master/json_formatter.go#L33). It would be nice to be able to rename `msg` to `message`.

**Comment by PurpureGecko** [comment]
There is option to modify fields name: you can implement Formatter interface for your log format and set it to Logger.

**Comment by dqminh** [comment]
That's what we do now, but it feels pretty inconvenient just to change one field name.

**Comment by stuart-warren** [comment]
There is a logstash json formatter:
```
import (
    log "github.com/Sirupsen/logrus"
    "github.com/Sirupsen/logrus/formatters/logstash"
)
...
log.SetFormatter(&logstash.LogstashFormatter{})
...
```

**Comment by aarongreenlee** [comment]
Thanks for bringing this up. Configuration of the field names seems reasonable to me. Would you be open to submitting a pull-request to provide this behavior with some test cases to exercise it, @dqminh?

**Comment by dqminh** [comment]
@aarongreenlee done, PTAL at #383

**Comment by dmathieu** [comment]
This has been fixed in #396.
